### PR TITLE
Added action for pytest-cov

### PR DESCRIPTION
**Pull request by ope-adejumo**
**Description**

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by aerorahul** [comment]
This is great @ope-adejumo 
I would make a suggestion:
- after the `cov-html` directory is created, perhaps create an artifact and make it available for the user to download it (if they desire) See the https://github.com/NOAA-EMC/hpc-stack/blob/develop/.github/workflows/build_ubuntu.yaml `Upload logs` part of the runner.  

It would be nice to view the cov-html on the web for `develop`.  Perhaps we can use the `gh-pages` functionality to view the code coverage.  What do you and @ryanlong1004 think?

File: .github/workflows/pytest-cov.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        run: |
+          pip install pytest-cov
+
+      - name: Run pytest-cov
+        run: |
+          pytest --cov-config=.coveragerc
+                 --cov-report html:cov_html
+                 --cov-fail-under 50
+                 --cov=src/uwtools
+                 workflow-tools/tests/
```
Comment (christinaholtNOAA) [inline .github/workflows/pytest-cov.yaml:39]: Does this maybe need to be run from inside the clone to get a pass?

**Review by christinaholtNOAA — CHANGES_REQUESTED** [review]

File: .github/workflows/pytest-cov.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install dependencies
+        run: |
+          pip install --upgrade pip
+
+      - name: Install pytest-cov
```
Comment (ryanlong1004) [inline .github/workflows/pytest-cov.yaml:29]: Any dependencies we need should exist in setup.cfg.

https://github.com/ufs-community/workflow-tools/blob/32aafc7d5d29714b09a8966f19583dd25d213403/setup.cfg#L52

`pytest-cov` already exists.  You can install `dev` dependencies using:

```pip install -e .[dev]```

File: .github/workflows/pytest-cov.yaml (hunk 1 above)
Comment (ryanlong1004) [reply .github/workflows/pytest-cov.yaml:39]: * change `--cov-fail-under  50` to fail under 5.  that's a sticky topic.
* remove `workflow-tools/tests/` the `tests` directory is already configured in `setup.py/setup.cfg`

File: .coveragerc [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[run]
```
Comment (ryanlong1004) [inline .coveragerc:1]: Delete this file and update the github action.

**Review by ryanlong1004 — CHANGES_REQUESTED** [review]

File: .github/workflows/pytest-cov.yaml (hunk 1 above)
Comment (ope-adejumo) [reply .github/workflows/pytest-cov.yaml]: changes implemented. workflow now passes

**Review by ryanlong1004 — APPROVED** [review]

**Comment by ryanlong1004** [comment]
> This is great @ope-adejumo I would make a suggestion:
> 
> * after the `cov-html` directory is created, perhaps create an artifact and make it available for the user to download it (if they desire) See the https://github.com/NOAA-EMC/hpc-stack/blob/develop/.github/workflows/build_ubuntu.yaml `Upload logs` part of the runner.
> 
> It would be nice to view the cov-html on the web for `develop`. Perhaps we can use the `gh-pages` functionality to view the code coverage. What do you and @ryanlong1004 think?

@aerorahul see #54

**Review by christinaholtNOAA — APPROVED** [review]

**Pull request merged**